File: simple-stack-example-basic-fragment/src/main/java/com/zhuinden/navigationexamplefrag/DashboardKey.java
package com.zhuinden.navigationexamplefrag;

import com.google.auto.value.AutoValue;

/**
 * Created by Owner on 2017. 06. 29..
 */
@AutoValue
public abstract class DashboardKey extends BaseKey {
    public static DashboardKey create() {
        return new AutoValue_DashboardKey();
    }

    @Override
    protected BaseFragment createFragment() {
        return new DashboardFragment();
    }
}

File: simple-stack-example-kotlin/src/main/kotlin/com/zhuinden/simplestackexamplekotlin/DashboardKey.kt
package com.zhuinden.simplestackexamplekotlin

import android.os.Parcelable
import paperparcel.PaperParcel

/**
 * Created by Owner on 2017.11.13.
 */

@PaperParcel
data class DashboardKey(val tag: String = DashboardKey::class.java.name) : BaseKey() {
    // using data class makes you not need to implement hashCode/equals/toString
    // the other option is using `object` where you MUST override `toString`
    override fun createFragment(): BaseFragment = DashboardFragment()

    companion object {
        @JvmField val CREATOR: Parcelable.Creator<DashboardKey> = PaperParcelDashboardKey.CREATOR
    }
}
